tests/test_bin_trie.py

```diff
@@ -36,3 +41,44 @@ def test_bin_trie_different_order_insert(k, v, random):
         for k, v in kv_pairs:
             trie.delete(k)
         assert trie.root_hash == BLANK_HASH
+
+
+def test_bin_trie_delete_subtrie():
```

Comment:
Can you refactor this test to use `pytest.mark.parametrize` so that we don't end up with any accidental leakage between the different test cases.